src/data/releases/release.model.ts

```diff
@@ -22,61 +22,83 @@ import { createHash } from 'crypto';
 import { FilterQuery } from 'mongodb';
 import mongoose from 'mongoose';
 
+/**
+ * `CREATED` -> `CALCULATING` -> `CALCULATED` -> `BUILDING` -> `BUILT` -> `PUBLISHING` -> `PUBLISHED`
+ */
 export enum ReleaseState {
-  ACTIVE = 'ACTIVE',
-  LOCKED = 'LOCKED',
+  CREATED = 'CREATED',
+
+  CALCULATING = 'CALCULATING',
+  CALCULATED = 'CALCULATED',
+  ERROR_CALCULATE = 'ERROR_CALCULATE',
+
+  BUILDING = 'BUILDING',
+  BUILT = 'BUILT',
+  ERROR_BUILD = 'ERROR_BUILD',
+
+  PUBLISHING = 'PUBLISHING',
   PUBLISHED = 'PUBLISHED',
+  ERROR_PUBLISH = 'ERROR_PUBLISH',
 }
 
 export interface Release {
   _id: string;
-  version: string;
+  version?: string;
   state: ReleaseState;
+  error?: string;
 
-  calculatedAt: Date;
   filesKept: string[];
   filesAdded: string[];
   filesRemoved: string[];
 
-  publishedAt: Date;
+  calculatedAt?: Date;
+  builtAt?: Date;
+  publishedAt?: Date;
   indices: string[];
-  label: string;
-  snapshot: string;
+  label?: string;
+  snapshot?: string;
 }
```

Comment:
Corrections made throughout this data model to track the properties that are optional, as many of these are not set until later in the release process.